app/scripts/lib/nonce-tracker.js

```diff
@@ -4,11 +4,17 @@ const Mutex = require('await-semaphore').Mutex
 
 class NonceTracker {
 
-  constructor ({ provider, getPendingTransactions, getConfirmedTransactions }) {
+  constructor ({
```

Comment:
not a blocker but this is getting a little long maybe not now but replace with config and destructure on the line underneath